x/dualstaking/keeper/grpc_query_provider_delegators.go

```diff
@@ -0,0 +1,43 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/x/dualstaking/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) ProviderDelegators(goCtx context.Context, req *types.QueryProviderDelegatorsRequest) (*types.QueryProviderDelegatorsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	nextEpoch, err := k.getNextEpoch(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var delegations []types.Delegation
+	indices := k.delegationFS.GetAllEntryIndicesWithPrefix(ctx, req.Provider)
+	for _, ind := range indices {
+		var delegation types.Delegation
+		found := k.delegationFS.FindEntry(ctx, ind, nextEpoch, &delegation)
+		if !found {
+			delegator, provider, chainID := types.DelegationKeyDecode(ind)
+			utils.LavaFormatError("critical: delegationFS entry index has no entry", fmt.Errorf("provider delegation not found"),
```

Comment:
need not be critical:  it is ok for an index to exist but not the delegation, for example if was deleted then the index will remain around for some time (even though the entry is dead, i.e. otherwise invisible)

Reply:
done. see https://github.com/lavanet/lava/commit/991a0e74722061319cbef1cd3199f4f893b87702